File: src/main/java/io/curiousoft/ijudi/ordermanagent/model/BasketItem.java
package io.curiousoft.ijudi.ordermanagent.model;

public class BasketItem {

    private String name;
    private int quantity;
    private double price;
    private double discountPerc;

    public BasketItem(String name, int quantity, double price, double discountPerc) {
        this.name = name;
        this.quantity = quantity;
        this.price = price;
        this.discountPerc = discountPerc;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public double getDiscountPerc() {
        return discountPerc;
    }

    public void setDiscountPerc(double discountPerc) {
        this.discountPerc = discountPerc;
    }
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/model/BasketItem.kt
package io.curiousoft.izinga.commons.model

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

class BasketItem(@JsonProperty("name") var name: String,
                 @JsonProperty("quantity") var quantity: Int,
                 @JsonProperty("price") var price: Double,
                 @JsonProperty("discountPerc") var discountPerc: Double) {

    @JsonIgnore
    var customerId: String? = null
    var externalUrl: String? = null
    var storePrice = 0.0
    var options: List<SelectionOption>? = null

    // weight in kilograms for this single item (default 0.0)
    var weight: Double = 0.0
    var width: Double = 0.0
    var length: Double = 0.0
    var height: Double = 0.0

    val totalPrice: Double
        get() = price * quantity

    val discountedPrice: Double
    get() = -1 * totalPrice * discountPerc


    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as BasketItem
        return name == that.name
    }
}